### PR TITLE
move the cancellation token out of Current

### PR DESCRIPTION
**Pull request by pepone**
This PR moves the dispatch cancellation token out of Current, it is now the las param for dispatch methods

### CODE REVIEW

File: csharp/src/Ice/IObject.cs [hunk 1]
```diff
@@ -13,10 +14,15 @@ public interface IObject
         /// <summary>Dispatches a request on this servant.</summary>
         /// <param name="request">The <see cref="IncomingRequestFrame"/> to dispatch.</param>
         /// <param name="current">Holds decoded header data and other information about the current request.</param>
+        /// <param name="cancel">A cancellation token that is notified of cancellation when the dispatch is cancelled.
```
Comment (bernardnormier) [inline csharp/src/Ice/IObject.cs:17]: I would use the American spelling for canceled, with one L.

File: csharp/src/Ice/IceDiscovery/Lookup.cs [hunk 2]
```diff
@@ -41,7 +46,7 @@ public void FindAdapterById(string domainId, string adapterId, ILookupReplyPrx?
                 try
                 {
                     Debug.Assert(reply != null);
-                    reply.FoundAdapterByIdAsync(adapterId, proxy, isReplicaGroup);
+                    reply.FoundAdapterByIdAsync(adapterId, proxy, isReplicaGroup, cancel: cancel);
```
Comment (bernardnormier) [inline csharp/src/Ice/IceDiscovery/Lookup.cs:49]: You're using a named parameter here to skip the defaulted context parameter?

**Review by bernardnormier — APPROVED** [review]
Looks good.

File: csharp/src/Ice/IceDiscovery/Lookup.cs (hunk 2 above)
Comment (pepone) [reply csharp/src/Ice/IceDiscovery/Lookup.cs:49]: yes

File: csharp/src/Ice/IObject.cs (hunk 1 above)
Comment (pepone) [reply csharp/src/Ice/IObject.cs:17]: ok

**Pull request merged**